lib/utils/visUtil.cpp

```diff
@@ -238,6 +238,97 @@ double movingAverage::average() {
     return current_value;
 }
 
+sampleBuffer::sampleBuffer(size_t size) :
+    rbuf(std::make_unique<double[]>(size)),
+    front(0),
+    end(0),
+    buf_size(size),
+    count(0){};
+
+void sampleBuffer::add_sample(double sample) {
+    rbuf[end] = sample;
+    if (count < buf_size) {
+        count++;
+    }
+
+    if (count == buf_size) {
+        front = (front + 1) % buf_size;
+    }
+
+    end = (end + 1) % buf_size;
+}
+
+double sampleBuffer::get_max() {
+    double max = std::numeric_limits<double>::min();
+    size_t index;
+
+    if (count == 0) {
+        return 0.0;
+    }
+
+    for (size_t i = 0; i < count; i++) {
+        index = (front + i) % buf_size;
+        if (max < rbuf[index]) {
+            max = rbuf[index];
+        }
+    }
+
+    return max;
+}
+
+double sampleBuffer::get_min() {
+    double min = std::numeric_limits<double>::max();
+    size_t index;
+
+    if (count == 0) {
+        return 0.0;
+    }
+
+    for (size_t i = 0; i < count; i++) {
+        index = (front + i) % buf_size;
+        if (min > rbuf[index]) {
+            min = rbuf[index];
+        }
+    }
+
+    return min;
+}
+
+double sampleBuffer::get_avg() {
+    double sum = 0.0;
+    size_t index;
+
+    if (count == 0) {
+        return 0.0;
+    }
+
+    for (size_t i = 0; i < count; i++) {
+        index = (front + i) % buf_size;
+        sum += rbuf[index];
+    }
+
+    return sum / count;
+}
+
+double sampleBuffer::get_std_dev() {
+    double standardDeviation = 0.0;
+    double mean;
+    size_t index;
+
+    if (count == 0) {
+        return 0.0;
+    }
+
+    mean = this->get_avg();
+
+    for (size_t i = 0; i < count; i++) {
+        index = (front + i) % buf_size;
+        standardDeviation += pow(rbuf[index] - mean, 2);
+    }
+
+    return sqrt(standardDeviation / count);
```

Comment:
For most of what we are measuring using the sample standard deviation will make more sense.
```suggestion
    return sqrt(standardDeviation / (count - 1));
```